internal/provisioner/kops_provisioner.go

```diff
@@ -233,12 +255,85 @@ func (provisioner *KopsProvisioner) CreateCluster(cluster *model.Cluster) error
 		return err
 	}
 
+	// Old tiller pod is deleted because of the upgrade action and therefore wait is needed
+	logger.Infof("Waiting %d seconds for the old Tiller pod to be replaced...", wait)
+	time.Sleep(time.Duration(wait) * time.Second)
```

Comment:
Hmm, if we just wait 60 seconds here, is it possible that everything may not be ready?

Also, if it completes sooner than 60 seconds, we will wait the full time regardless. As you can see above, we developed helper functions to wait for something to happen and then return asap. We should do that here as well.

Reply:
@gabrieljackson I used the helper function initially. The problem with this is that when Helm inits the first time it creates a pod and after some configuration the upgrade command forces the deletion of this pod and creation of a new one. If you use the helper to wait for a pod creation sometimes you end up with it waiting for the terminating pod to come up and it fails. That is why I added a 60 second sleep as this is enough for the tiller pod to come up.

Reply:
I added a new check that checks the whole helm readiness. The reason is that even if we check the tiller pod and it moves to running state, tiller needs a couple of more seconds to be ready to accept commands from helm. By checking the whole helm status for up to a specific time we ensure that it moves forward only when really ready.